app/javascript/controllers/transition_controller.js

```diff
@@ -0,0 +1,35 @@
+import { Controller } from '@hotwired/stimulus'
+
+// Example:
+//
+// <div data-controller="transition" data-transition-toggle-class="!hidden">
+//   <div id="element-that-shows-and-hides" class="block" data-transition-target="me"></div>
+//   <a href="#" data-action="transition#toggleClass">click to toggle</a>
+// </div>
+//
+// Every element that is of target = me will have the class toggled.
+
+export default class extends Controller {
+  static classes = [ "toggle" ]
+  static targets = [ "me" ]
+  static values = {
+    afterTimeout: Number
+  }
+
+  connect() {
+    console.log(`connect`)
+    if (this.afterTimeoutValue) setTimeout(() => this.toggleClass(), this.afterTimeoutValue)
+  }
+
+  toggleClass() {
+    console.log('toggling', this.meTargets[0])
+    console.log('class', this.toggleClasses[0])
+    this.meTargets.forEach(element => {
+      element.classList.toggle(...this.toggleClasses)
+    })
+
+    // Showing and hiding elements can cause the page to flow differently, very similarly to what happens when the
+    // browser size changes. Throw this event in case we have other listeners on the resize event.
+    window.dispatchEvent(new Event('resize'))
```

Comment:
I see what you're saying, but do we have anything catching the resize event or is this preemptive optimization? I haven't had to listen for the resize event in ages because browser layout via CSS is so much more robust than what we used to do with dhtml layouts.

Reply:
Yes, right now the scroll position of the page is recalculated on browser resize. This ensures that (a) if I'm at the bottom of the page and I resize my browser, I stay at the bottom of the page and (b) if I'm scrolled up reading a specific message and I resize my browser, it keeps me at that message.

Currently, there are multiple circumstances **besides** browser resizing in which the browser height and thus the scroll position need to be recalculated. Right now I'm enabling those other stimulus controllers to "communicate" with the scroll position controller by simply throwing a resize event.

This particular one right here is because hiding the left-hand column needs to trigger a scroll position recalculate. Basically, even though the browser itself didn't change size the right may have just done so. The other places I'm dispatching this event are within composer since that thing grows & shrinks in size.

It's a little bit of an overload of the resize event. Actually, sheesh, I should just create a custom event for this. I'll do that in a follow-up PR.